src/Config/Converters/EntityRestOptionsConverter.cs

```diff
@@ -14,7 +14,7 @@ internal class EntityRestOptionsConverter : JsonConverter<EntityRestOptions>
     {
         if (reader.TokenType == JsonTokenType.StartObject)
         {
-            EntityRestOptions restOptions = new(Methods: EntityRestOptions.DEFAULT_SUPPORTED_VERBS, Path: null, Enabled: true);
+            EntityRestOptions restOptions = new(Methods: Array.Empty<SupportedHttpVerb>(), Path: null, Enabled: true);
```

Comment:
Per specs, the `Methods` property should be nullable. We might need to take this into consideration maybe in a subsequent PR.

Reply:
The non-nullability is probably more a result of my misunderstanding of what the role of `Methods` was than anything else.

It can probably be made nullable, it'd then just be a matter of finding the impact downstream in the codebase and having the appropriate `null` checks in place.

Reply:
I've created [this issue](https://github.com/Azure/data-api-builder/issues/1571) to track this refactoring.